### ISSUE TITLE
Problems with uploading mapping.txt file for de-obfuscation 

### ISSUE THREAD

**Issue by Maxwell6635** [DataDog/dd-sdk-android-gradle-plugin#117]
Hi!

I have a problem with uploading mapping.txt for de-obfuscation to DataDog it will always hitting network error I try switch multiple different ISP or machine also the same.

**Describe what happened**
 ```
Task :app:uploadMappingBetaRelease FAILED
Error uploading the mapping file for com.boostorium.beta:5.23.0 {variant:beta}
java.net.ConnectException: Failed to connect to sourcemap-intake.datadoghq.eu/2600:1901:0:606b:0:0:0:0:443
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:297)
	at okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207)
	at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226)
	at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106)
	at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74)
	at okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201)
	at okhttp3.internal.connection.RealCall.execute(RealCall.kt:154)
	at com.datadog.gradle.plugin.internal.OkHttpUploader.upload(OkHttpUploader.kt:59)
	at com.datadog.gradle.plugin.DdMappingFileUploadTask.applyTask(DdMappingFileUploadTask.kt:153)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:58)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:51)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:29)
	at org.gradle.api.internal.tasks.execution.TaskExecution$2.run(TaskExecution.java:239)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:68)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeAction(TaskExecution.java:224)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeActions(TaskExecution.java:207)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeWithPreviousOutputFiles(TaskExecution.java:190)
	at org.gradle.api.internal.tasks.execution.TaskExecution.execute(TaskExecution.java:168)
	at org.gradle.internal.execution.steps.ExecuteStep.executeInternal(ExecuteStep.java:89)
	at org.gradle.internal.execution.steps.ExecuteStep.access$000(ExecuteStep.java:40)
	at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:53)
	at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:50)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:50)
	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:40)
	at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:68)
	at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:38)
	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:48)
	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:36)
	at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:41)
	at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:74)
	at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:55)
	at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:51)
	at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:29)
	at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.execute(CaptureStateAfterExecutionStep.java:61)
	at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.execute(CaptureStateAfterExecutionStep.java:42)
	at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:60)
	at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:27)
	at org.gradle.internal.execution.steps.BuildCacheStep.executeWithoutCache(BuildCacheStep.java:188)
	at org.gradle.internal.execution.steps.BuildCacheStep.lambda$execute$1(BuildCacheStep.java:75)
	at org.gradle.internal.Either$Right.fold(Either.java:175)
	at org.gradle.internal.execution.caching.CachingState.fold(CachingState.java:59)
	at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:73)
	at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:48)
	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:38)
	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:27)
	at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:36)
	at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:22)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:109)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$2(SkipUpToDateStep.java:56)
	at java.base/java.util.Optional.orElseGet(Optional.java:369)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:56)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:38)
	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:73)
	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:44)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:37)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:27)
	at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:89)
	at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:50)
	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:114)
	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:57)
	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:76)
	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:50)
	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.lambda$execute$2(SkipEmptyWorkStep.java:93)
	at java.base/java.util.Optional.orElseGet(Optional.java:369)
	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:93)
	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:34)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:38)
	at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:43)
	at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:31)
	at org.gradle.internal.execution.steps.AssignWorkspaceStep.lambda$execute$0(AssignWorkspaceStep.java:40)
	at org.gradle.api.internal.tasks.execution.TaskExecution$3.withWorkspace(TaskExecution.java:284)
	at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:40)
	at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:30)
	at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:37)
	at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:27)
	at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:44)
	at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:33)
	at org.gradle.internal.execution.impl.DefaultExecutionEngine$1.execute(DefaultExecutionEngine.java:76)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:142)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:131)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:74)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:402)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:389)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:382)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:368)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:61)
	at java.base/java.lang.Thread.run(Thread.java:829)
	Suppressed: okhttp3.internal.http2.StreamResetException: stream was reset: NO_ERROR
		at okhttp3.internal.http2.Http2Stream.checkOutNotClosed$okhttp(Http2Stream.kt:646)
		at okhttp3.internal.http2.Http2Stream$FramingSink.emitFrame(Http2Stream.kt:557)
		at okhttp3.internal.http2.Http2Stream$FramingSink.write(Http2Stream.kt:532)
		at okio.ForwardingSink.write(ForwardingSink.kt:29)
		at okhttp3.internal.connection.Exchange$RequestBodySink.write(Exchange.kt:218)
		at okio.RealBufferedSink.emitCompleteSegments(RealBufferedSink.kt:255)
		at okio.RealBufferedSink.writeAll(RealBufferedSink.kt:194)
		at okhttp3.RequestBody$Companion$asRequestBody$1.writeTo(RequestBody.kt:167)
		at okhttp3.MultipartBody.writeOrCountBytes(MultipartBody.kt:157)
		at okhttp3.MultipartBody.writeTo(MultipartBody.kt:93)
		at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.kt:59)
		at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
		at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:34)
		... 135 more
Caused by: java.net.ConnectException: No route to host (connect failed)
	at java.base/java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:399)
	at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:242)
	at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:224)
Caused by: java.net.ConnectException: No route to host (connect failed)

	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.base/java.net.Socket.connect(Socket.java:609)
	at okhttp3.internal.platform.Platform.connectSocket(Platform.kt:128)
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:295)
	... 141 more
```

**Steps to reproduce the issue:**
Execution failed for task ':app:uploadMappingBetaRelease'.

**Describe what you expected:**
No error happen

**Additional context**
Android Studio Dolphin | 2021.3.1
Build #AI-213.7172.25.2113.9014738, built on September 1, 2022
Runtime version: 11.0.13+0-b1751.21-8125866 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 12.6
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 16
Registry:
    external.system.auto.import.disabled=true
    ide.text.editor.with.preview.show.floating.toolbar=false

Non-Bundled Plugins:
    com.mroche.JsonHelper (1.1.2)
    org.sonarlint.idea (7.0.0.52289)
    mobi.hsz.idea.gitignore (4.4.0)
    Docker (213.7172.6)

**Comment by Maxwell6635** [comment]
ad0398@AD0398-ADE-MB-01 ~ % curl -I https://sourcemap-intake.datadoghq.com
HTTP/2 400 
date: Tue, 27 Sep 2022 05:59:32 GMT
content-type: application/json
content-length: 83
cross-origin-resource-policy: cross-origin
x-content-type-options: nosniff
strict-transport-security: max-age=15724800;

**Comment by xgouchet** [comment]
Hi @Maxwell6635 , which version of the Plugin are you using?

**Comment by xgouchet** [comment]
Also can you check if you have a firewall or proxy that might prevent the upload request from succeeding? The `No route to host` error message seems to indicate an issue on the network layer, which we don't have control on in our plugin.

**Comment by Maxwell6635** [comment]
@xgouchet 
Hi ,

here are the plugins version that currently in used.

plugins {
    id "de.timfreiheit.resourceplaceholders" version "0.4"
    id("com.datadoghq.dd-sdk-android-gradle-plugin") version "1.4.0"
}

Also can you check if you have a firewall or proxy that might prevent the upload request from succeeding? The No route to host error message seems to indicate an issue on the network layer, which we don't have control on in our plugin.

Hmm, but when we do a ping to https://sourcemap-intake.datadoghq.com/ is able connect. Any idea for this ?

**Comment by xgouchet** [comment]
Hi, I tested the same plugin version as you're using and was able to upload a mapping file without any issue. 
Are you doing the ping from the same machine trying to upload the mapping file? Also the network connection is performed from within the JVM running your gradle task, so maybe there's some rule on the underlying OS preventing the network connection?

**Comment by Maxwell6635** [comment]
@xgouchet Hi, sorry for late reply. 
Are you doing the ping from the same machine trying to upload the mapping file? 
- Yes, ping from the same machine its seems able to connect. But when running gradle task from Android Studio it will hit the issues.
- 
Also the network connection is performed from within the JVM running your gradle task, so maybe there's some rule on the underlying OS preventing the network connection?
- Not really, we tested with or without our internal vpn service it also facing the same issues. 
- Even the VPN is on, we still able to ping to https://sourcemap-intake.datadoghq.com/

**Comment by xgouchet** [comment]
Can you try running the gradle task on a different computer/different network to check if the issue is comming from our plugin or the environment
If the issue persists with a different computer and different network, can you send us your whole Gradle setup so that we can try reproducing the issue locally

**Comment by 0xnm** [comment]
Hello @Maxwell6635! I hope the issue is resolved for you now, so I'm closing it due to inactivity. Feel free to re-open it if needed.

**Comment by stevoland** [comment]
Have run into this on a circleci executor. I can ssh to the machine and ping `sourcemap-intake.datadoghq.eu` fine. The task runs fine on my local machine.

@0xnm Is there any other way to manually upload the mapping? With datadog-ci?

**Comment by 0xnm** [comment]
Hello @stevoland! Do you have exactly the same exception as in the original post? Can you please share the plugin version you are using? I would also check if you can establish simple HTTP connection with our intake from CircleCI executor.

**Comment by stevoland** [comment]
Hi @0xnm, hmm, seems it is a bit different in this regard:

```
Uploading mapping file for native:2.70.0 {variant:staging} (site=datadoghq.eu):

Creating request with GZIP encoding.
Error uploading the mapping file for native:2.70.0 {variant:staging}
java.net.ConnectException: Failed to connect to sourcemap-intake.datadoghq.eu/2600:1901:0:606b:0:0:0:0:443
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:297)
	at okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207)
[...]
Caused by: java.net.ConnectException: Network is unreachable (connect failed)
	at java.base/java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:412)
	at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:255)
	at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:237)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.base/java.net.Socket.connect(Socket.java:609)
	at okhttp3.internal.platform.Platform.connectSocket(Platform.kt:128)
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:295)
``` 

Full stacktrace [in this gist](https://gist.github.com/stevoland/3d3b8653f936776fc26a59c5f92fb8f0)

From `android/app/build.gradle`
```
plugins {
    id("com.datadoghq.dd-sdk-android-gradle-plugin") version "1.8.+"
}
```

From the executor:
```
$ curl -I https://sourcemap-intake.datadoghq.eu/2600:1901:0:606b:0:0:0:0:443
HTTP/2 404
content-type: application/json
content-length: 81
cross-origin-resource-policy: cross-origin
accept-encoding: identity,gzip,x-gzip,deflate,x-deflate,zstd
x-content-type-options: nosniff
strict-transport-security: max-age=31536000; includeSubDomains; preload
date: Tue, 25 Apr 2023 10:11:48 GMT
via: 1.1 google
alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
```

CircleCI executor config:
```
executors:
  android:
    resource_class: large
    machine:
      image: android:2022.08.1
```

Any ideas greatly appreciated!

**Comment by 0xnm** [comment]
Hello @stevoland!

The address you see in the stacktrace can be broken in the following pieces:

* Hostname - `sourcemap-intake.datadoghq.eu`
* IPv6 representation of the host - `2600:1901:0:606b:0:0:0:0`
* Port number - `443`.

So to test connectivity with `curl` you can simply use `curl -I https://sourcemap-intake.datadoghq.eu/` (because `https` will point to port 443).

But anyway, from your test I can see that connectivity can be established (because you've got HTTP response back).

Can you also tell us what is the version of `OkHttp` resolved in the build environment? You can see it by running `gradlew buildEnvironment` task.

Do you also have this issue all the time or it happens occasionally?

**Comment by stevoland** [comment]
Hey @0xnm 

I've added the output of `gradlew buildEnvironment` [to the gist](https://gist.github.com/stevoland/3d3b8653f936776fc26a59c5f92fb8f0). I don't see any okhttp, oddly. I've also added (most of) the `app/build.gradle` if that helps

> Do you also have this issue all the time or it happens occasionally?

Every time

Many thanks for your help

**Comment by 0xnm** [comment]
@stevoland Sorry, I wasn't clear enough: you need to run `buildEnvironment` task for the module where you have Datadog plugin applied, in your case if it is called `app` module, then it will be `gradlew :app:buildEnvironment`.

**Comment by stevoland** [comment]
@0xnm 

```
> Task :app:buildEnvironment

------------------------------------------------------------
Project ':app'
------------------------------------------------------------

classpath
+--- io.sentry.android.gradle:io.sentry.android.gradle.gradle.plugin:2.1.5
|    \--- io.sentry:sentry-android-gradle-plugin:2.1.5
|         \--- org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.4.32 -> 1.5.31
|              +--- org.jetbrains.kotlin:kotlin-stdlib:1.5.31 -> 1.8.10
|              |    +--- org.jetbrains.kotlin:kotlin-stdlib-common:1.8.10
|              |    \--- org.jetbrains:annotations:13.0
|              \--- org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.5.31
|                   \--- org.jetbrains.kotlin:kotlin-stdlib:1.5.31 -> 1.8.10 (*)
\--- com.datadoghq.dd-sdk-android-gradle-plugin:com.datadoghq.dd-sdk-android-gradle-plugin.gradle.plugin:1.8.+ -> 1.8.0
     \--- com.datadoghq:dd-sdk-android-gradle-plugin:1.8.0
          +--- org.jetbrains.kotlin:kotlin-stdlib:1.8.10 (*)
          +--- com.squareup.okhttp3:okhttp:4.10.0
          |    +--- com.squareup.okio:okio:3.0.0
          |    |    \--- com.squareup.okio:okio-jvm:3.0.0
          |    |         +--- org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.5.31 (*)
          |    |         \--- org.jetbrains.kotlin:kotlin-stdlib-common:1.5.31 -> 1.8.10
          |    \--- org.jetbrains.kotlin:kotlin-stdlib:1.6.20 -> 1.8.10 (*)
          \--- org.json:json:20180813
```

**Comment by 0xnm** [comment]
Hello @stevoland. It turns out that this issue happens because of the [HTTP/2 handling bug](https://github.com/square/okhttp/commit/e46a200fceaa5fbd9c9278eebe9c978399bd329c) in the OkHttp library. Good news is that bug is fixed in the recent version 4.11.0 of OkHttp.

We will release a new plugin version with OkHttp 4.11.0, but until it is not yet there, you can simply force OkHttp 4.11.0 usage by adding `classpath("com.squareup.okhttp3:okhttp:4.11.0")` into your `buildscript > dependencies` block (and then you can verify that this version is applied by running `gradlew :app:buildEnvironment`.

By looking on the bug description it seems that it happens when there is some issue with request from the server perspective, so if you face it and won't be able to figure out by yourself, please open a different ticket.

**Comment by stevoland** [comment]
@0xnm Your suggestion worked for me. Thanks so much for your help
